rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/nodesource/NodeSourceConfigurationParser.java

```diff
@@ -0,0 +1,140 @@
+    }
+
+    public NodeSourceConfiguration parseNodeSourceConfiguration(String json) throws ImportException {
+
+        JSONObject jsonObject;
+
+        try {
+            jsonObject = this.controller.parseJSON(json).isObject();
+        } catch (JSONException e) {
+            throw new ImportException("The imported node source is not a valid JSON file", e);
+        }
+
+        try {
+            String nodeSourceName = jsonObject.get("nodeSourceName").isString().stringValue();
+            boolean nodesRecoverable = jsonObject.get("nodesRecoverable").isBoolean().booleanValue();
+
+            JSONObject infrastructurePluginDescriptorJson = jsonObject.get("infrastructurePluginDescriptor").isObject();
+            String infrastructurePluginName = infrastructurePluginDescriptorJson.get("pluginName")
+                                                                                .isString()
+                                                                                .stringValue();
+            PluginDescriptor infrastructurePluginDescriptor = getPluginDescriptor(infrastructurePluginDescriptorJson,
+                                                                                  infrastructurePluginName);
+
+            JSONObject policyPluginDescriptorJson = jsonObject.get("policyPluginDescriptor").isObject();
+            String policyPluginName = policyPluginDescriptorJson.get("pluginName").isString().stringValue();
+            PluginDescriptor policyPluginDescriptor = getPluginDescriptor(policyPluginDescriptorJson, policyPluginName);
+
+            return new NodeSourceConfiguration(nodeSourceName,
+                                               nodesRecoverable,
+                                               infrastructurePluginDescriptor,
+                                               policyPluginDescriptor);
+        } catch (RuntimeException e) {
+            throw new ImportException("The imported node source has incorrect parameters.", e);
+        }
+    }
+
+    public HashMap<String, PluginDescriptor> parsePluginDescriptors(String json) {
+
+        JSONArray arr = this.controller.parseJSON(json).isArray();
+        HashMap<String, PluginDescriptor> plugins = new HashMap<>();
+
+        for (int i = 0; i < arr.size(); i++) {
+            JSONObject p = arr.get(i).isObject();
+
+            String pluginName = p.get("pluginName").isString().stringValue();
+            PluginDescriptor pluginDescriptor = getPluginDescriptor(p, pluginName);
+
+            plugins.put(pluginName, pluginDescriptor);
+        }
+
+        return plugins;
+    }
+
+    private PluginDescriptor getPluginDescriptor(JSONObject p, String pluginName) {
+
+        String pluginDescription = p.get("pluginDescription").isString().stringValue();
+        PluginDescriptor desc = new PluginDescriptor(pluginName, pluginDescription);
+
+        JSONArray fields = p.get("configurableFields").isArray();
+        for (int j = 0; j < fields.size(); j++) {
```

Comment:
can we streaminize this as well ?

Reply:
It does not bring much clarity as I need to use code blocs in the stream.